app/main.py

```diff
@@ -1,3 +1,86 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return F"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(km=self.km + other)
+        else:
+            raise TypeError(
+                f"isinstance({type(other)}, (Distance, int, float)) is True")
+
+    def __iadd__(self, other: "Distance") -> "Distance":
```

Comment:
`other` can be int and float too. Fix it everywhere
```suggestion
    def __iadd__(self, other: int | float | "Distance") -> "Distance":
```